### PR TITLE
Using sahara with both Swift and HDFS as backend

### PR DESCRIPTION
**Pull request by tellesnobrega**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,50 +1,52 @@
-from setuptools import setup, find_packages
```
Comment (armstrongmsg) [inline setup.py:1]: It shows as if the whole file has been modified. Any idea why?

File: application_manager/utils/remote.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+
+
+def write_file_to(remote_file, data, run_as_root=False, timeout=120):
```
Comment (armstrongmsg) [inline application_manager/utils/remote.py:19]: Not implemented?

File: application_manager/service/api/v10.py [hunk 3]
```diff
@@ -25,12 +25,12 @@
 
 
 def execute(data):
-    authorization = authorizer.get_authorization(api.authorization_url,
-                                                 data['bigsea_username'],
-                                                 data['bigsea_password'])
-    if not authorization['success']:
-        return 'Error: Authentication failed. User not authorized'
-    
+    #authorization = authorizer.get_authorization(api.authorization_url,
```
Comment (armstrongmsg) [inline application_manager/service/api/v10.py:28]: Commented code

File: application_manager/service/api/__init__.py [hunk 4]
```diff
@@ -1,55 +1,56 @@
...
+project_id = config.get('os-generic', 'project_id')
+auth_ip = config.get('os-generic', 'auth_ip')
+
+user = config.get('os-generic', 'user')
+password = config.get('os-generic', 'password')
+domain = config.get('os-generic', 'user_domain_name')
+log_path = config.get('os-generic', 'log_path')
+
+container = config.get('spark-sahara', 'swift_container')
+#user_domain_name = config.get('spark-sahara', 'user_domain_name')
```
Comment (armstrongmsg) [inline application_manager/service/api/__init__.py:41]: Commented code

File: application_manager/plugins/sahara/plugin.py [hunk 5]
```diff
@@ -121,15 +126,14 @@ def start_application(self, data, spark_applications_ids, app_id):
             pred_cluster_size = optimizer.get_cluster_size(api.optimizer_url,
                                                            hosts)
 
+            pred_cluster_size = req_cluster_size
```
Comment (armstrongmsg) [inline application_manager/plugins/sahara/plugin.py]: Is this line correct? Should it assign this value here?

File: application_manager/plugins/sahara/plugin.py [hunk 6]
```diff
@@ -377,6 +305,209 @@ def _create_cluster(self, sahara, connector, req_cluster_size,
...
+                                                    job_type)
+
+        LOG.log("%s | Starting job..." % (time.strftime("%H:%M:%S")))
+
+        # Running job
+        configs = os_utils.get_job_config(connector, plugin,
+                                          cluster_size, user, password,
+                                          args, main_class)
+
+        job = connector.create_job_execution(sahara, job_template_id,
```
Comment (armstrongmsg) [inline application_manager/plugins/sahara/plugin.py:308]: In the case of swift paths, are we still going to use sahara jobs?

Comment (betobetodev) [reply application_manager/plugins/sahara/plugin.py:308]: Yes. If hdfs, use standalone job execution. If swift, use sahara job execution.

**Pull request merged**